src/llfs/bloom_filter.hpp

```diff
@@ -22,39 +22,66 @@
 #include <batteries/static_assert.hpp>
 #include <batteries/suppress.hpp>
 
-#include <boost/functional/hash.hpp>
+#include <xxhash.h>
 
 #include <cmath>
+#include <type_traits>
 
 namespace llfs {
 
 struct BloomFilterParams {
   usize bits_per_item;
 };
 
+namespace bloom_filter_impl {
```

Comment:
I put that overload set in its own namespace because it is implementation detail (so I suppose it could live in `namespace detail` as we use elsewhere).  Basically, I just want to warn off any downstream usage of these APIs, so that future changes won't cause compatibility breaks.

Reply:
Renamed the namespace to `detail` to be more consistent with existing conventions.